packages/tax/src/services/tax-module-service.ts

```diff
@@ -166,4 +167,145 @@ export default class TaxModuleService<
   ) {
     return await this.taxRateRuleService_.create(data, sharedContext)
   }
+
+  async getTaxLines(
+    items: (TaxTypes.TaxableItemDTO | TaxTypes.TaxableShippingDTO)[],
+    calculationContext: TaxTypes.TaxCalculationContext,
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<(TaxTypes.ItemTaxLineDTO | TaxTypes.ShippingTaxLineDTO)[]> {
+    const regions = await this.taxRegionService_.list(
+      {
+        $or: [
+          {
+            country_code: calculationContext.address.country_code,
+            province_code: null,
+          },
+          {
+            country_code: calculationContext.address.country_code,
+            province_code: calculationContext.address.province_code,
+          },
+        ],
+      },
+      {},
+      sharedContext
+    )
+
+    const toReturn = await promiseAll(
+      items.map(async (item) => {
+        const isShipping = "shipping_option_id" in item
+        let ruleQuery = isShipping
+          ? [
+              {
+                reference: "shipping_option",
+                reference_id: item.shipping_option_id,
+              },
+            ]
+          : [
+              {
+                reference: "product",
+                reference_id: item.product_id,
+              },
+              {
+                reference: "product_type",
+                reference_id: item.product_type_id,
+              },
+            ]
+
+        const rates = await this.taxRateService_.list(
+          {
+            $and: [
+              { tax_region_id: { $in: regions.map((r) => r.id) } },
+              { $or: [{ is_default: true }, { rules: { $or: ruleQuery } }] },
+            ],
+          },
+          { relations: ["rules", "tax_region"] },
+          sharedContext
+        )
+
+        const prioritizedRates = this.prioritizeRates(rates, item)
+
+        const rate = prioritizedRates[0]
+        const toReturn = {
+          rate_id: rate.id,
+          rate: rate.rate,
+          code: rate.code,
+          name: rate.name,
+        }
+
+        if (isShipping) {
+          return [{ shipping_line_id: item.id, ...toReturn }]
+        }
+
+        return [{ line_item_id: item.id, ...toReturn }]
```

Comment:
Why do we return an array, just to flatten it afterwards?

Reply:
Some rates may be combinable - this is coming in a future PR :)